infrastructure/src/test/java/org/corfudb/infrastructure/management/failuredetector/FailureDetectorTestContext.java

```diff
@@ -0,0 +1,157 @@
+
+    private ClusterStateTestContext clusterStateContext;
+
+    public FailureDetectorTestContext(String localEndpoint) {
+        this.localEndpoint = localEndpoint;
+
+        fdDataStoreBuilder
+                .localEndpoint(localEndpoint)
+                .dataStore(mock(DataStore.class));
+
+        advisor = new CompleteGraphAdvisor(localEndpoint);
+        fsAdvisor = new FileSystemAdvisor();
+
+        FailureDetectorDataStore fdDatastore = fdDataStoreBuilder.build();
+
+        healingAgent = HealingAgent.builder()
+                .dataStore(fdDatastore)
+                .advisor(advisor)
+                .fsAdvisor(fsAdvisor)
+                .failureDetectorWorker(fdWorkerMock)
+                .localEndpoint(localEndpoint)
+                .runtimeSingleton(runtimeMock);
+
+        failuresAgent = FailuresAgent.builder()
+                .fdDataStore(fdDatastore)
+                .advisor(advisor)
+                .fsAdvisor(fsAdvisor)
+                .localEndpoint(localEndpoint)
+                .runtimeSingleton(runtimeMock);
+
+        clusterStateContext = new ClusterStateTestContext(localEndpoint);
+    }
+
+    public void setupDataStore(Consumer<FailureDetectorDataStore.FailureDetectorDataStoreBuilder> setupTask) {
+        setupTask.accept(fdDataStoreBuilder);
+    }
+
+    public void setupHealingAgent(Consumer<HealingAgentBuilder> setupTask) {
+        setupTask.accept(healingAgent);
+    }
+
+    public HealingAgent getHealingAgentSpy() {
+        return spy(healingAgent.build());
+    }
+
+    public FailuresAgent getFailuresAgentSpy() {
+        return spy(failuresAgent.build());
+    }
+
+    public ClusterState getClusterState() {
+        return clusterStateContext.getClusterState();
+    }
+
+    public PollReport getPollReportMock(ClusterState clusterState) {
+        PollReport pollReportMock = mock(PollReport.class);
+        when(pollReportMock.getClusterState()).thenReturn(clusterState);
+        return pollReportMock;
+    }
+
+    public void setupClusterState(Consumer<ClusterStateTestContext> task) {
+        task.accept(clusterStateContext);
+    }
+
+    public Layout getLayoutMock() {
+        return mock(Layout.class);
+    }
+
+    public static class ClusterStateTestContext {
+        private ClusterState.ClusterStateBuilder state = ClusterState.builder();
+        private Map<String, NodeState> nodes = new HashMap<>();
+        @Getter
+        private FileSystemStats fsStats;
+
+        public ClusterStateTestContext(String localEndpoint) {
+            state.localEndpoint(localEndpoint);
+            fsStats = getFileSystemStats();
+
+            long epoch = 0;
+
+            nodes.put(NodeNames.A, nodeState(NodeNames.A, epoch, Optional.of(fsStats), OK, OK, OK));
+            nodes.put(NodeNames.B, nodeState(NodeNames.B, epoch, Optional.of(fsStats), OK, OK, OK));
+            nodes.put(NodeNames.C, nodeState(localEndpoint, epoch, Optional.of(fsStats), OK, OK, OK));
```

Comment:
Should we replace `NodeNames.C ` to match `localEndpoint` here? I also noticed that `localEndpoint` has the value `NodeNames.B` in FailuresAgentTest. Maybe we can assert here that `localEndpoint` is distinct from A and B?

Reply:
done